### PR TITLE
docs/ipsec: misc improvements

### PR DESCRIPTION
**Pull request by kaworu**
<del>This PR is based on top of https://github.com/cilium/cilium/pull/15975</del>

### CODE REVIEW

**Review by qmonnet — APPROVED** [review]

File: Documentation/gettingstarted/encryption-ipsec.rst [hunk 1]
```diff
@@ -17,24 +17,25 @@ as Cilium-managed host traffic, will be encrypted using IPsec. This guide uses
 Kubernetes secrets to distribute keys. Alternatively, keys may be manually
 distributed, but that is not shown here.
 
-Packets destined to the same node they were sent out of are not encrypted.
-This is an intended behavior as it would not provide any benefits because the
-raw traffic on the node can be seen.
+Packets are sent not encrypted when they are destined to the same node from
+which they were sent. This behavior is intended. Encryption would provide no
+benefits in that case, given that the raw traffic can be observed on the node
+anyway.
```
Comment (rolinh) [inline Documentation/gettingstarted/encryption-ipsec.rst]: If you update this paragraph, could you please update the corresponding one in the wireguard guide?

File: Documentation/gettingstarted/encryption-ipsec.rst [hunk 2]
```diff
@@ -156,20 +158,20 @@ Validate the Setup
 ==================
 
 Run a ``bash`` shell in one of the Cilium pods with
-``kubectl -n kube-system exec -ti <cilium pod> -- bash`` and execute the
-following commands:
+``kubectl -n kube-system exec -ti ds/cilium -- bash`` and execute the following
+commands:
```
Comment (rolinh) [inline Documentation/gettingstarted/encryption-ipsec.rst:161]: Would you mind doing this change in the wireguard guide as well?

File: Documentation/gettingstarted/encryption-ipsec.rst (hunk 1 above)
Comment (kaworu) [reply Documentation/gettingstarted/encryption-ipsec.rst]: This part of the diff is from https://github.com/cilium/cilium/pull/15975

File: Documentation/gettingstarted/encryption-ipsec.rst (hunk 2 above)
Comment (kaworu) [reply Documentation/gettingstarted/encryption-ipsec.rst:161]: It's already `ds/cilium` in the [wireguard guide](https://github.com/kaworu/cilium/blob/pr/kaworu/docs/encryption-ipsec/Documentation/gettingstarted/encryption-wireguard.rst#validate-the-setup).

**Review by rolinh — APPROVED** [review]

**Comment by qmonnet** [comment]
https://github.com/cilium/cilium/pull/15975 has been merged.

File: Documentation/gettingstarted/encryption-ipsec.rst [hunk 3]
```diff
@@ -197,16 +197,18 @@ To replace cilium-ipsec-keys secret with a new key:
     data=$(echo "{\"stringData\":{\"keys\":\"$((($KEYID+1))) "rfc4106\(gcm\(aes\)\)" $(echo $(dd if=/dev/urandom count=20 bs=1 2> /dev/null| xxd -p -c 64)) 128\"}}")
     kubectl patch secret -n kube-system cilium-ipsec-keys -p="${data}" -v=1
 
-Then restart Cilium agents to transition to the new key. During transition the
+Then restart Cilium agents to transition to the new key with
+``kubectl delete pod -n kube-system -l k8s-app=cilium`` During transition the
```
Comment (qmonnet) [inline Documentation/gettingstarted/encryption-ipsec.rst]:
```suggestion
``kubectl delete pod -n kube-system -l k8s-app=cilium``. During transition the
```

**Review by qmonnet — CHANGES_REQUESTED** [review]

**Review by qmonnet — APPROVED** [review]

**Pull request merged**